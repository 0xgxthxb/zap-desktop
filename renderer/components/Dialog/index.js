export DialogAppCrashed from './DialogAppCrashed'
export DialogLndCrashed from './DialogLndCrashed'
