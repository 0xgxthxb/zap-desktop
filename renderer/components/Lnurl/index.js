export LnurlAuthPrompt from './LnurlAuthPrompt'
export LnurlChannelPrompt from './LnurlChannelPrompt'
