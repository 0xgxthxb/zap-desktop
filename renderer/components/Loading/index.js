import LoadingBolt from './Loading'

export default LoadingBolt
