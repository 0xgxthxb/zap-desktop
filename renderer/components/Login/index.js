import Login from './Login'

export LoginNotAllowed from './LoginNotAllowed'
export default Login
