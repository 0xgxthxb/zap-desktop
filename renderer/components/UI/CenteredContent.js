import React from 'react'
import { Flex } from 'rebass'

const CenteredContent = props => (
  <Flex
    alignItems="center"
    flexDirection="column"
    height="100%"
    justifyContent="center"
    width={1}
    {...props}
  />
)

export default CenteredContent
