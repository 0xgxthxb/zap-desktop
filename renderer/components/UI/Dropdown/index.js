import Dropdown from './Dropdown'

export { DropdownButton } from './Dropdown'
export default Dropdown
