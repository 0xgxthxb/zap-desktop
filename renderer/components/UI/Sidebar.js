import React from 'react'
import PropTypes from 'prop-types'
import { Flex } from 'rebass/styled-components'

const Sidebar = ({ hasShadow, ...rest }) => (
  <Flex
    as="aside"
    color="primaryText"
    flexDirection="column"
    width={4 / 12}
    {...rest}
    sx={{
      overflow: 'hidden',
      boxShadow: hasShadow ? 'm' : null,
      zIndex: 1,
    }}
  />
)

Sidebar.propTypes = {
  hasShadow: PropTypes.bool,
}

Sidebar.defaultProps = {
  hasShadow: true,
}

Sidebar.small = props => <Sidebar {...props} width={3 / 16} />
Sidebar.medium = props => <Sidebar {...props} width={4 / 16} />
Sidebar.large = props => <Sidebar {...props} width={6 / 16} />

Sidebar.small.displayName = 'Sidebar Small'
Sidebar.medium.displayName = 'Sidebar Medium'
Sidebar.large.displayName = 'Sidebar Large'

export default Sidebar
