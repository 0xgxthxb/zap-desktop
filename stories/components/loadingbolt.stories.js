import React from 'react'
import { storiesOf } from '@storybook/react'
import { boolean } from '@storybook/addon-knobs'
import { Page } from 'components/UI'
import Loading from 'components/LoadingBolt'

storiesOf('Components', module).addWithChapters('LoadingBolt', {
  subtitle: 'Animation to indicate application is loading.',
  chapters: [
    {
      sections: [
        {
          title: 'Bolt',
          sectionFn: () => {
            const isLoading = boolean('Is loading', true)
            return (
              <Page>
                <Loading hasClose isLoading={isLoading} onClose={() => {}} variant="bolt" />
              </Page>
            )
          },
        },
        {
          title: 'App',
          sectionFn: () => {
            const isLoading = boolean('Is loading', true)
            return (
              <Page>
                <Loading hasClose isLoading={isLoading} onClose={() => {}} variant="app" />
              </Page>
            )
          },
        },
        {
          title: 'Launchpad',
          sectionFn: () => {
            const isLoading = boolean('Is loading', true)
            return (
              <Page>
                <Loading isLoading={isLoading} variant="launchpad" />
              </Page>
            )
          },
        },
      ],
    },
  ],
})
