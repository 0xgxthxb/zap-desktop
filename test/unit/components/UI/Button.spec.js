import React from 'react'
import { Button } from 'components/UI'
import { renderWithTheme } from '@zap/test/unit/__helpers__/renderWithTheme'

describe('component.UI.Button', () => {
  it('should render correctly', () => {
    const tree = renderWithTheme(<Button>Click Me</Button>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
