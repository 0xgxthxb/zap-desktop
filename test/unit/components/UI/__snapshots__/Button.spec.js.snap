// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component.UI.Button should render correctly 1`] = `
.c0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: primary;
  border: 0;
  border-radius: 4px;
  font-family: Roboto,system-ui,sans-serif;
  background-color: #313340;
  color: #fd9800;
  border-radius: 5px;
  padding-left: 45px;
  padding-right: 45px;
  padding-top: 16px;
  padding-bottom: 16px;
}

.c0:hover:enabled {
  background-color: #353745;
}

.c0:focus {
  background-color: #353745;
}

.c2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 13px;
  font-weight: 400;
}

.c1 {
  -webkit-transition: all 0.25s;
  transition: all 0.25s;
  outline: none;
  font-weight: 300;
  line-height: '18px';
  white-space: nowrap;
  text-align: center;
}

.c1:disabled {
  opacity: 0.5;
}

.c1:hover:enabled {
  cursor: pointer;
}

<button
  className="c0 c1"
>
  <div
    className="c2"
    fontSize="m"
    fontWeight="normal"
  >
    Click Me
  </div>
</button>
`;
